src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java

```diff
@@ -265,14 +265,10 @@ public Converter() {
   @Option(
       name = "experimental_repository_cache_urls_as_default_canonical_id",
```

Comment:
I wonder should we just remove this flag directly? So that users depending on this feature actually get an error and can migrate to the new solution. Or is there a warning for deprecated flag?

Reply:
I added a deprecation warning:
```
WARNING: Option 'experimental_repository_cache_urls_as_default_canonical_id' is deprecated: This behavior is enabled by default for http_* and jvm_* rules and no longer controlled by this flag. Use --repo_env=BAZEL_HTTP_RULES_URLS_AS_DEFAULT_CANONICAL_ID=0 to disable it instead.
```